### PR TITLE
b2c wishlist-items endpoint tests

### PR DESCRIPTION
**Pull request by gorav-sharma-metacube**
CC-16471 B2C Wishlist-items-endpoints

### CODE REVIEW

File: tests/api/b2c/glue/wishlist_items/positive.robot [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    When I send a POST request:    /wishlists/${wishlist_id}/wishlist-items    {"data": {"type": "wishlist-items","attributes": {"sku": "${product_sku_1}"}}}
+    Then Response status code should be:    201 
+    And Response reason should be:    Created
+    And Save value to a variable:    [data][id]    wishlist_items_id
+    And Response body parameter should not be EMPTY:    [data][attributes][sku]
+    And Response body parameter should be:    [data][type]    wishlist-items
+    And Response body parameter should be:    [data][id]    ${wishlist_items_id}
+    And Response body parameter should be:    [data][attributes][sku]    ${product_sku_1}
+    And Response body parameter should be:    [data][attributes][id]    ${wishlist_items_id}
+    And Response body parameter should be:    [data][attributes][availability]    None
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/positive.robot:24]: there are fields availability and price array - they need to be checked and they need to be not empty- they need to have the values of the product with this sku. if they are empty we still need to write a test that checks that they are not empty - it will fail because that is a bug

File: tests/api/b2c/glue/wishlist_items/positive.robot [hunk 2]
```diff
@@ -0,0 +1,64 @@
+*** Settings ***
+Suite Setup        SuiteSetup
+Test Setup        TestSetup 
+Resource    ../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
+#Post
+Adding_item_in_wishlist
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/positive.robot:8]: we also need a test which adds several products to wishlist - 2 variants of the same abstract product

File: tests/api/b2c/glue/wishlist_items/positive.robot [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    Then Response status code should be:    201 
+    And Response reason should be:    Created
+    And Save value to a variable:    [data][id]    wishlist_items_id
+    And Response body parameter should not be EMPTY:    [data][attributes][sku]
+    And Response body parameter should be:    [data][type]    wishlist-items
+    And Response body parameter should be:    [data][id]    ${wishlist_items_id}
+    And Response body parameter should be:    [data][attributes][sku]    ${product_sku_1}
+    And Response body parameter should be:    [data][attributes][id]    ${wishlist_items_id}
+    And Response body parameter should be:    [data][attributes][availability]    None
+    [Teardown]    Run Keywords    I send a DELETE request:    /wishlists/${wishlist_id}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/positive.robot:25]: we also need to check selflink

File: tests/api/b2c/glue/wishlist_items/positive.robot [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    ...    AND    Save value to a variable:    [data][id]    wishlist_id
+    When I send a POST request:    /wishlists/${wishlist_id}/wishlist-items    {"data": {"type": "wishlist-items","attributes": {"sku": "${product_sku_4}"}}}
+    Then Response status code should be:    201 
+    And Response reason should be:    Created
+    And Save value to a variable:    [data][id]    wishlist_items_id
+    And Response body parameter should be:    [data][type]    wishlist-items
+    And Response body parameter should be:    [data][id]    ${wishlist_items_id}
+    And Response body parameter should be:    [data][attributes][sku]    ${product_sku_4}
+    And Response body parameter should be:    [data][attributes][id]    ${wishlist_items_id}
+    And Response body parameter should be:    [data][attributes][availability]    None
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/positive.robot:45]: why are we doing all these checks in a delete test? we do not need these checks

File: tests/api/b2c/glue/wishlist_items/positive.robot [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+     [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}" } }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_id
+    When I send a POST request:    /wishlists/${wishlist_id}/wishlist-items    {"data": {"type": "wishlist-items","attributes": {"sku": "${product_sku_4}"}}}
+    Then Response status code should be:    201 
+    And Response reason should be:    Created
+    And Save value to a variable:    [data][id]    wishlist_items_id
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/positive.robot:40]: adding wishlist item should be a part of setup, not steps

File: tests/api/b2c/glue/wishlist_items/positive.robot [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    And Response body parameter should be:    [data][type]    wishlist-items
+    And Response body parameter should be:    [data][id]    ${wishlist_items_id}
+    And Response body parameter should be:    [data][attributes][sku]    ${product_sku_4}
+    And Response body parameter should be:    [data][attributes][id]    ${wishlist_items_id}
+    And Response body parameter should be:    [data][attributes][availability]    None
+    When I send a DELETE request:    /wishlists/${wishlist_id}/wishlist-items/${product_sku_4}
+    Then Response status code should be:    204
+    And Response reason should be:    No Content
+    When I send a GET request:    /wishlists/${wishlist_id}/wishlist-items/${product_sku_4}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/positive.robot:49]: why are you doing this check? this should be done in a negative test already - please remove it

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+*** Test Cases ***
+#Post
+
+Adding_item_in_wishlist_by_invalid_Access_Token
+     [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_reference_id
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:15]: you do not need all these steps - error with invalid access token will happen even if there are no item or wishlist, please remove the unneeded steps to speed up the test

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+    ...  AND    I send a DELETE request:    /wishlists/${wishlist_reference_id}
+    ...  AND    Response status code should be:    204
+    
+Adding_item_in_wishlist_by_without_Access_Token
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_reference_id
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:32]: same here - all this preparation is unneeeded because system doe sot get to check the wishlist - it will throw correct error without this preparation

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+    ...  AND    Response status code should be:    204
+
+Adding_item_with_invalid_sku
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_reference_id
+    When I send a POST request:   /wishlists/${wishlist_reference_id}/wishlist-items    {"data": {"type": "wishlist-items","attributes": {"sku": "${product_sku_2}"}}}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:50]: invalid sku should be hardcoded

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+    ...    AND    Response status code should be:    204
+
+Adding_item_with_deactivated_item_sku
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_reference_id
+    When I send a POST request:   /wishlists/${wishlist_reference_id}/wishlist-items    {"data": {"type": "wishlist-items","attributes": {"sku": "${product_sku_2}"}}}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:65]: why is invalid sku and deactivated sku are he same? this is not deactivated, the value here is invalid - what do you check here and why?

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+     Response status code should be:    422
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:66]: 'Then' is missing

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+    ...    AND    Response status code should be:    204
+
+#Delete
+Deleting_item_in_wishlist_by_invalid_Access_Token
+     [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_reference_id
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:80]: we do not need this preparation, it is not used by the test - result will be the same without it

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+    ...  AND    I send a DELETE request:    /wishlists/${wishlist_reference_id}
+    ...  AND    Response status code should be:    204
+    
+Deleting_item_in_wishlist_by_without_Access_Token
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_reference_id
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:97]: we do not need this preparation, it is not used by the test - result will be the same without it

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 14]
```diff
@@ -0,0 +1,161 @@
+*** Settings ***
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:1]: please fix folder structure
it should be: wishlist_endpoits/wishlist_items - and there should be your positive and negative files

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+Suite Setup        SuiteSetup
+Test Setup        TestSetup 
+Resource    ../../../../../resources/common/common_api.robot
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/negative.robot:4]: please add these to settings and add them in future tests
*** Settings ***

Default Tags glue

File: tests/api/b2c/glue/wishlist_items/positive.robot [hunk 16]
```diff
@@ -0,0 +1,64 @@
+*** Settings ***
+Suite Setup        SuiteSetup
+Test Setup        TestSetup 
+Resource    ../../../../../resources/common/common_api.robot
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_items/positive.robot:4]: please add these to settings and add them in future tests
*** Settings ***

Test Setup TestSetup
Default Tags glue

File: resources/environments/environments_api_b2c.json [hunk 17]
```diff
@@ -2,7 +2,7 @@
     "environment": "api_b2c",
 
     "env": {
-        "glue_url": "http://glue.de.spryker.local",
+        "glue_url": "https://glue.de.b2c.internal-testing.demo-spryker.com",
```
[the remark this answers is not in the document]
Comment (gorav-sharma-metacube) [reply resources/environments/environments_api_b2c.json]: Fixed

File: resources/environments/environments_api_b2c.json [hunk 18]
```diff
@@ -250,6 +250,14 @@
         "qty_of_categories_in_category_trees": "6",
         "qty_of_subcategories_in_category_trees": "2",
         "color_name": "black"
+    },
+    "wishlist":{
+        "product_sku_1":  "056_31714843",
+        "product_sku_2":  "swewewewew",
```
[the remark this answers is not in the document]
Comment (gorav-sharma-metacube) [reply resources/environments/environments_api_b2c.json]: Fixed

File: resources/environments/environments_api_b2c.json [hunk 19]
```diff
@@ -250,6 +250,14 @@
...
+        "product_sku_3": "1000_11",
+        "product_sku_4":    "191_17681791",
+        "product_sku_5":    "",
```
[the remark this answers is not in the document]
Comment (gorav-sharma-metacube) [reply resources/environments/environments_api_b2c.json]: Fixed

File: resources/environments/environments_api_b2c.json [hunk 20]
```diff
@@ -250,6 +250,14 @@
         "qty_of_categories_in_category_trees": "6",
         "qty_of_subcategories_in_category_trees": "2",
         "color_name": "black"
+    },
+    "wishlist":{
```
[the remark this answers is not in the document]
Comment (gorav-sharma-metacube) [reply resources/environments/environments_api_b2c.json]: Fixed

File: tests/api/b2c/glue/wishlist_items/positive.robot (hunk 1 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/positive.robot:24]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 11 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:66]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 12 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:80]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 13 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:97]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+    ...  AND    Response status code should be:    204
+
+Deleting_item_in_wishlist_with_empty_sku
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_reference_id
+     When I send a DELETE request:    /wishlists/${wishlist_reference_id}/wishlist-items/${product_sku_5}
```
[the remark this answers is not in the document]
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:115]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot [hunk 22]
```diff
@@ -0,0 +1,161 @@
+*** Settings ***
+Suite Setup        SuiteSetup
+Test Setup        TestSetup 
+Resource    ../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
```
[the remark this answers is not in the document]
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:6]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 14 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:1]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 15 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:4]: Fixed

File: tests/api/b2c/glue/wishlist_items/positive.robot (hunk 16 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/positive.robot:4]: Fixed

File: tests/api/b2c/glue/wishlist_items/positive.robot (hunk 2 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/positive.robot:8]: Fixed

File: tests/api/b2c/glue/wishlist_items/positive.robot (hunk 3 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/positive.robot:25]: Fixed

File: tests/api/b2c/glue/wishlist_items/positive.robot (hunk 4 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/positive.robot:45]: Fixed

File: tests/api/b2c/glue/wishlist_items/positive.robot (hunk 5 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/positive.robot:40]: Fixed

File: tests/api/b2c/glue/wishlist_items/positive.robot (hunk 6 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/positive.robot:49]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 8 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:32]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 9 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:50]: Fixed

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 10 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot:65]: Fixed

File: resources/environments/environments_api_b2c.json [hunk 23]
```diff
@@ -87,29 +75,23 @@
...
         "alternatives_label": "Alternatives available",
-
         "category_id_lvl1": "11",
         "category_items_qty_lvl1": "39",
         "category_id_lvl2": "12",
         "category_items_qty_lvl2": "38",
         "category_id_lvl3": "13",
         "category_items_qty_lvl3": "28",
         "category_tree_branches_qty": "6",
-
```
Comment (OKrivtsova) [inline resources/environments/environments_api_b2c.json:104]: why did you remove new lines all over the file? it makes file harder to read. you will also have issues merging it, please revert

File: resources/environments/environments_api_b2c.json [hunk 24]
```diff
@@ -139,53 +117,47 @@
...
         "concrete_product_multiple_image_set": "158_29885222",
         "concrete_product_multiple_image_set_name": "Asus ZenPad Z170C",
         "concrete_available_product_with_stock": "001_25904006",
-        "concrete_available_product_without_stock": "212_123"
-            
+        "concrete_available_product_without_stock": "212_123",
+        "abstract_product_with_multiple_variants": "130",
+        "concrete_product_id_first": "130_29285281",
+        "concrete_product_id_second": "130_24326086",
+        "concrete_product_id_third": "008_30692992"
```
Comment (OKrivtsova) [inline resources/environments/environments_api_b2c.json:157]: please rename concrete_product_id_first to concrete_product_sku_multivariant_variant1 and so on for others

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/positive.robot [hunk 25]
```diff
@@ -0,0 +1,63 @@
...
+    ...    AND    Response status code should be:    204
+
+Adding_multiple_variant_of_abstract_product_in_wishlist
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}" } }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]        wishlist_id
+    When I send a POST request:    /wishlists/${wishlist_id}/wishlist-items    {"data": {"type": "wishlist-items","attributes": {"sku": "${concrete_product_id_first}"}}}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_endpoints/wishlist_items/positive.robot:39]: why don't we check success of the first post?

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/positive.robot [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+    AND I send a POST request:    /wishlists/${wishlist_id}/wishlist-items    {"data": {"type": "wishlist-items","attributes": {"sku": "${concrete_product_id_second}"}}}
+    Then Response status code should be:    201 
+    And Response reason should be:    Created
+    And Save value to a variable:    [data][id]    wishlist_items_id
+    And Response body parameter should not be EMPTY:    [data][attributes][sku]
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_endpoints/wishlist_items/positive.robot:44]: we should GET wishlist with included items to check that there are really 2 items in it

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot [hunk 27]
```diff
@@ -0,0 +1,264 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+#Post
+
+Adding_item_in_wishlist_by_invalid_Access_Token
+     [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:10]: I get access token for the customer:    ${yves_user_email} - this is still getting a token which we do not need

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot [hunk 28]
```diff
@@ -0,0 +1,264 @@
...
+    ...    AND    I set Headers:    Authorization=3485h7
+    When I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    Then Response status code should be:    401
+    And Response reason should be:    Unauthorized
+    And Response should return error code:    001
+    And Response should return error message:    Invalid access token.
+    
+Adding_item_in_wishlist_by_without_Access_Token
+        [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:19]: I get access token for the customer:    ${yves_user_email} - this is still getting a token which we do not need

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot [hunk 29]
```diff
@@ -0,0 +1,264 @@
...
+    And Response should return error message:    "Cant find wishlist."
+
+Adding_item_with_deactivated_item_sku
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}
+    ...    AND    I send a POST request:    /wishlists    {"data": { "type": "wishlists","attributes": { "name": "${wishlist_name}"} }}
+    ...    AND    Response status code should be:    201 
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    wishlist_reference_id
+    When I send a POST request:   /wishlists/${wishlist_reference_id}/wishlist-items    {"data": {"type": "wishlist-items","attributes": {"sku": "${concrete_product_id_third}"}}}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:125]: please since this product is deactivated, please add some information about it being deactivated to variable name (I asked you to update names above, this should be done in scope of that)
if a product is special we should always reflect it in variable name - it is easier to transfer to other envs that way

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 14 above)
Comment (OKrivtsova) [reply tests/api/b2c/glue/wishlist_items/negative.robot:1]: now we have duplicated files - please make sure there are no duplicated ad all changes ad fixes are in the remaining files
image

File: tests/api/b2c/glue/wishlist_items/positive.robot (hunk 16 above)
Comment (OKrivtsova) [reply tests/api/b2c/glue/wishlist_items/positive.robot:4]: Test Setup TestSetup is ow missing. please add both default tags AND test setup to files
image

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot [hunk 30]
```diff
@@ -0,0 +1,264 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:4]: same here - Test Setup is missing

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 15 above)
Comment (OKrivtsova) [reply tests/api/b2c/glue/wishlist_items/negative.robot:4]: now tags are added but TEst Setup is missing

File: tests/api/b2c/glue/wishlist_items/negative.robot (hunk 14 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_items/negative.robot]: Fixed

File: resources/environments/environments_api_b2c.json (hunk 23 above)
Comment (gorav-sharma-metacube) [reply resources/environments/environments_api_b2c.json:104]: Fixed

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/positive.robot (hunk 25 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/positive.robot]: Fixed

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/positive.robot (hunk 26 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/positive.robot:47]: Fixed

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot (hunk 27 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot]: Fixed

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot (hunk 28 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot]: Fixed

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot (hunk 29 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot]: Fixed

File: resources/environments/environments_api_b2c.json (hunk 24 above)
Comment (gorav-sharma-metacube) [reply resources/environments/environments_api_b2c.json]: Fixed

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot [hunk 31]
```diff
@@ -0,0 +1,261 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Test Setup     TestSetup
+Default Tags    glue
+
+*** Test Cases ***
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:7]: some tests are failing against 
https://glue.de.b2c.internal-testing.demo-spryker.com - please fix
image
image

File: tests/api/b2b/glue/cart_endpoints/carts/positive.robot [hunk 32]
```diff
@@ -283,15 +283,15 @@ Update_cart_with_existing_name
     And Response body parameter should be:    [data][attributes][priceMode]    ${gross_mode}
     And Response body parameter should be:    [data][attributes][currency]    ${currency_code_eur}
     And Response body parameter should be:    [data][attributes][store]    ${store_de}
-    And Response body parameter should NOT be:    [data][attributes][name]    "My Cart"
+    And Response body parameter should NOT be:   [data][attributes][name]    "My Cart"
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/cart_endpoints/carts/positive.robot:286]: why did you change this file? please revert it you should not touch carts

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot (hunk 31 above)
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:7]: Please refer to the attached screenshot. All cases are working fine at my end.
image

Comment (OKrivtsova) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:7]: wha about positive tests

Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:7]: Hello, please check screenshot for the positive test cases. "Adding item in wishlist" test case is not passing because of the API issue(Refer to bug id: CC-16555).
image

Comment (OKrivtsova) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:7]: please check tests against https://glue.de.b2c.internal-testing.demo-spryker.com/

Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:7]: Fixed

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot [hunk 33]
```diff
@@ -114,21 +114,6 @@ Adding_items_in_wishlist_by_another_customer_wishlist
     And Response should return error code:    201
     And Response should return error message:    "Cant find wishlist."
 
-Adding_item_with_deactivated_item_sku
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot]: why did you delete this test? there should be deactivated products

Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot]: Code reverted but there is no demo data there So i have added a comment there for no demo data.

File: tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot [hunk 34]
```diff
@@ -0,0 +1,246 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Test Setup     TestSetup
+Default Tags    glue
+
+*** Test Cases ***
+#Post
+
+Adding_item_in_wishlist_by_invalid_Access_Token
```
[the remark this answers is not in the document]
Comment (gorav-sharma-metacube) [reply tests/api/b2c/glue/wishlist_endpoints/wishlist_items/negative.robot:12]: Enabler Testsetup added. But after merging the code from master there was no changes regarding deactivated product variables. So i have mentioned the comment in the code.

File: resources/common/common_api.robot [hunk 35]
```diff
@@ -11,6 +11,9 @@ Library    ../../resources/libraries/common.py
 
 *** Variables ***
 # *** SUITE VARIABLES ***
+${current_url}                 http://glue.de.spryker.local
```
Comment (OKrivtsova) [inline resources/common/common_api.robot:14]: revert all changes to this file! you did not merge master correctly - you should not change this file in this PR!

Comment (gorav-sharma-metacube) [reply resources/common/common_api.robot:14]: fixed

Comment (OKrivtsova) [reply resources/common/common_api.robot:14]: not fixed. I still see changes in this file - this file should not have changes. please merge master again and make sure this file is overwritten with the changes from the master and not your changes

Comment (gorav-sharma-metacube) [reply resources/common/common_api.robot]: fixed

**Review by OKrivtsova — APPROVED** [review]
